pkg/chart/chart.go

```diff
@@ -326,28 +328,35 @@ func (t *Testing) InstallChart(chart string, valuesFiles []string) TestResult {
 
 	result := TestResult{Chart: chart}
 
-	if len(valuesFiles) > 0 {
-		for _, valuesFile := range valuesFiles {
-			release, namespace := util.CreateInstallParams(chart, t.config.BuildId)
-
-			defer t.kubectl.DeleteNamespace(namespace)
-			defer t.helm.DeleteRelease(release)
-			defer t.PrintPodDetailsAndLogs(namespace)
+	// Test with defaults if no values files are specified.
+	if len(valuesFiles) == 0 {
+		valuesFiles = append(valuesFiles, "")
+	}
 
-			if err := t.helm.InstallWithValues(chart, valuesFile, namespace, release); err != nil {
-				result.Error = err
-				break
-			}
-		}
-	} else {
+	for _, valuesFile := range valuesFiles {
+		releaseSelector := ""
 		release, namespace := util.CreateInstallParams(chart, t.config.BuildId)
-
-		defer t.kubectl.DeleteNamespace(namespace)
 		defer t.helm.DeleteRelease(release)
-		defer t.PrintPodDetailsAndLogs(namespace)
 
-		if err := t.helm.Install(chart, namespace, release); err != nil {
+		if t.config.Namespace != "" {
+			namespace = t.config.Namespace
+			releaseSelector = fmt.Sprintf("%s=%s", t.config.ReleaseLabel, release)
+		} else {
+			defer t.kubectl.DeleteNamespace(namespace)
```

Comment:
The problem now is that the cleanup order is no longer correct because you call this after `defer t.helm.DeleteRelease(release)`. Please keep the old order.